modules/mobile/app/controllers/mobile/v0/travel_pay_claims_controller.rb

```diff
@@ -0,0 +1,113 @@
+        rescue Faraday::ClientError, Faraday::ServerError => e
+          raise Common::Exceptions::InternalServerError, exception: e
+        end
+        claim = Mobile::V0::TravelPayClaimSummary.new({
+                                                        id: submitted_claim['claimId'],
+                                                        claimNumber: '',
+                                                        claimStatus: submitted_claim['status'].underscore.humanize,
+                                                        appointmentDateTime: params['appointment_date_time'],
+                                                        facilityId: params['facility_station_number'],
+                                                        facilityName: '',
+                                                        totalCostRequested: 0,
+                                                        reimbursementAmount: 0,
+                                                        createdOn: submitted_claim['createdOn'],
+                                                        modifiedOn: submitted_claim['modifiedOn']
+                                                      })
+        render json: TravelPayClaimSummarySerializer.new(claim), status: :created
+      end
+
+      private
+
+      def validated_params
+        smoc_params = {
+          appointment_date_time: params['appointment_date_time'],
+          facility_station_number: params['facility_station_number'],
+          appointment_type: params['appointment_type'] || 'Other',
+          is_complete: params['is_complete'] || false
+        }
+        if params['appointment_name'].present? && !params['appointment_name'].empty?
+          smoc_params[:appointment_name] =
+            params['appointment_name']
+        end
+
+        @validated_params ||= Mobile::V0::Contracts::TravelPaySmoc.new.call(smoc_params)
+      end
+
+      def get_appt_or_raise
+        appt_params = {
+          'appointment_date_time' => validated_params[:appointment_date_time],
+          'facility_station_number' => validated_params[:facility_station_number],
+          'appointment_type' => validated_params[:appointment_type],
+          'is_complete' => validated_params[:is_complete]
+        }
+        if validated_params[:appointment_name].present?
+          appt_params['appointment_name'] =
+            validated_params[:appointment_name]
+        end
+        appt_not_found_msg = "No appointment found for #{appt_params['appointment_date_time']}"
+        Rails.logger.info(message: "SMOC transaction: Get appt by date time: #{appt_params['appointment_date_time']}")
+        appt = appts_service.find_or_create_appointment(appt_params)
+
+        if appt[:data].nil?
+          Rails.logger.error(message: appt_not_found_msg)
+          raise Common::Exceptions::ResourceNotFound, detail: appt_not_found_msg
+        end
+
+        appt[:data]['id']
+      end
+
+      def get_claim_id(appt_id)
```

Comment:
What do you think of this? So then when it's used, we know it _returns_ a claim id.

```suggestion
      def claim_id(appt_id)
```

Reply:
I feel like this is ok as is, since it describes what it's doing (reaching out and getting the claim ID from the external service). 🤔